src/d3a/d3a_core/sim_results/device_statistics.py

```diff
@@ -49,8 +50,19 @@ def _device_price_stats(cls, area_dict: Dict, subdict: Dict, core_stats, current
         area_core_trades = core_stats[area_dict['uuid']].get('trades', [])
         trade_price_list = []
         for t in area_core_trades:
-            if t['seller'] == area_dict['name'] or t['buyer'] == area_dict['name']:
-                trade_price_list.append(t['energy_rate'] / 100.0)
+            if t['seller'] == area_dict['name']:
+                if ConstSettings.IAASettings.MARKET_TYPE == 1:
+                    trade_rate = t['energy_rate'] / 100.0
+                else:
+                    trade_rate = (t['energy_rate'] - t['fee_price'] / t['energy']) / 100.0
+                trade_price_list.append(trade_rate)
+            elif t['buyer'] == area_dict['name']:
+                if ConstSettings.IAASettings.MARKET_TYPE == 1:
+                    trade_rate = (t['energy_rate'] + t['fee_price'] / t['energy']) / 100.0
+                else:
+                    trade_rate = t['energy_rate'] / 100.0
+                trade_price_list.append(trade_rate)
```

Comment:
Minor, but There is some duplication in code (including src/d3a/d3a_core/sim_results/market_price_energy_day.py) i suggest to create two small utility functions called `trade_rate_without_fee` and `trade_rate_with_fee`. What do you think?